testsuite/integration/src/test/java/org/keycloak/testsuite/broker/OIDCKeycloakServerBrokerUserAttributeTest.java

```diff
@@ -0,0 +1,133 @@
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.services.managers.RealmManager;
+import org.keycloak.testsuite.KeycloakServer;
+import org.keycloak.testsuite.rule.AbstractKeycloakRule;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+
+
+public class OIDCKeycloakServerBrokerUserAttributeTest extends AbstractIdentityProviderTest {
+
+    private static final int PORT = 8082;
+
+    private static Keycloak keycloak1;
+    private static Keycloak keycloak2;
+
+    @ClassRule
+    public static AbstractKeycloakRule samlServerRule = new AbstractKeycloakRule() {
+
+        @Override
+        protected void configureServer(KeycloakServer server) {
+            server.getConfig().setPort(PORT);
+        }
+
+        @Override
+        protected void configure(KeycloakSession session, RealmManager manager, RealmModel adminRealm) {
+            server.importRealm(getClass().getResourceAsStream("/broker-test/test-broker-realm-with-oidc-multi-value-user-attributes.json"));
+        }
+
+        @Override
+        protected String[] getTestRealms() {
+            return new String[]{"realm-with-oidc-identity-provider"};
+        }
+    };
+
+    @Override
+    protected String getProviderId() {
+        return "kc-oidc-idp";
+    }
+
+    @BeforeClass
+    public static void beforeClazz() {
+        keycloak1 = Keycloak.getInstance("http://localhost:8081/auth", "master", "admin", "admin", org.keycloak.models.Constants.ADMIN_CLI_CLIENT_ID);
+        keycloak2 = Keycloak.getInstance("http://localhost:8082/auth", "master", "admin", "admin", org.keycloak.models.Constants.ADMIN_CLI_CLIENT_ID);
+    }
+
+    @Test
+    public void canAccessMultiValueAttribute() throws Exception {
+        UserModel user = assertSuccessfulAuthentication(getIdentityProviderModel(), "test-user", "test-user@localhost", false);
+        assertThat(user.getAttribute("mobile"), containsInAnyOrder("617-666-7777", "617-666-7778"));
+    }
+
+    @Test
+    public void canAccessSingleValueAttribute() throws Exception {
+        UserModel user = authenticate(getIdentityProviderModel(), "test-user-1", true);
+        assertThat(user.getAttribute("mobile"), containsInAnyOrder("617-666-7777"));
+        assertThat(user.getFirstAttribute("mobile"), is("617-666-7777"));
+    }
+
+
+    @Test
+    public void nonExistingAttributeLookupReturnsAnEmptyList() {
+        UserModel user = assertSuccessfulAuthentication(getIdentityProviderModel(), "test-user-noemail", "new@email.com", true);
+        assertThat(user.getAttribute("mobile"), is(empty()));
+    }
+
+    @Test
+    public void canUpdateBrokeredUserAttribute() {
+        authenticate(getIdentityProviderModel(), "test-user-2", true);
+
+       /* UserModel userByEmail = this.session.users().getUserByUsername("test-user-2", getRealm());
```

Comment:
Here I don't know what to do in order to be able to update the claim and get the updated attributes from the context in the user attribute mapper. I have tried different combinations and looked at the other tests but without success.